src/util/rangelist.cpp

```diff
@@ -40,8 +41,9 @@ QList<int> parseRangeList(const QString& input) {
     }
 
     std::sort(intList.begin(), intList.end());
-    const auto end = std::unique(intList.begin(), intList.end());
-    intList.erase(end, intList.end());
+    const auto constLast = make_const_iterator(
+            intList, std::unique(intList.begin(), intList.end()));
+    constErase(&intList, constLast, intList.constEnd());
```

Comment:
Yes, Qt has followed with Qt6.

Reply:
Right, I'm saying we could avoid the workaround here by refactoring the function to use a `std::vector` instead of QList. Then its unaffected by the version transition.

Reply:
Yes sure. That will start a bigger refactoring.
QList is not like std::vector we should consider such changes along the best container for a specific task. Doing it alone to avoid a temporary workaround is not worth the work IMHO.

Reply:
> QList is not like std::vector we should consider such changes along the best container for a specific task.

Unrelated to the discussion at hand but please elaborate. The only difference I know of is the implicit sharing COW sharing.

Reply:
Yes that is one point. The other is that QList is like a vector of pointers. Every small type is stored directly. Bigger types are allocated on the heap.

Reply:
@daschuer Did you tried to use QVector or std::vector instead?